src/tools/dart2native.md

```diff
@@ -1,7 +1,105 @@
 ---
 title: dart2native
 description: Command-line tool for AOT-compiling Dart source code.
-toc: false
 ---
 
-Coming soon.
+Use the `dart2native` command to AOT (ahead-of-time) compile
+a Dart program to native x64 machine code.
+The output of `dart2native` is either
+a **standalone executable** (the default)
+or an **AOT snapshot** that you can run with the [`dartaotruntime` command][].
+The `dart2native` command is supported on Windows, macOS, and Linux.
+
+{{site.alert.note}}
+  To execute Dart code without AOT compiling it,
+  use [`dart`, the standalone Dart VM](/tools/dart-vm).
+{{site.alert.end}}
+
+
+## Creating standalone executables
+
+Here's an example of using `dart2native` to create a standalone executable:
+
+```terminal
+$ dart2native bin/main.dart -o bin/my_app
+```
+
+You can distribute and run that executable like you would
+any other executable file:
+
+```terminal
+$ cp bin/my_app ~/bin
+$ my_app
+```
+
+
+## Creating AOT snapshots
+
+To create an AOT snapshot, add `-o aot` to the command:
+
+```terminal
+$ dart2native bin/main.dart -o aot
```

Comment:
I'm confused about this command. Above we use the same flag to nominate the output file, yet here we are using it to define the type of compilation. I'd understand if the file extension defines the compilation style, e.g.

```bash
$ dart2native bin/main.dart -o bin/main.aot
```

Reply:
Oops, I overcorrected. It should be `-k`. Let me fix this.

Reply:
OK, fixed. ptal.

(Also, I filed https://github.com/dart-lang/sdk/issues/39154 and fixed https://github.com/dart-lang/site-www/issues/1948#issuecomment-535382470.)

Reply:
lgtm, thanks =)